packages/telemetry/src/flight-recorder.js

```diff
@@ -371,18 +366,23 @@ export const makeMemoryMappedCircularBuffer = async ({
 
 /**
  *
- * @param {Pick<Awaited<ReturnType<typeof makeMemoryMappedCircularBuffer>>, 'writeCircBuf'>} circBuf
+ * @param {Pick<Awaited<ReturnType<typeof makeSimpleCircularBuffer>>, 'writeCircBuf'>} circBuf
  */
 export const makeSlogSenderFromBuffer = ({ writeCircBuf }) => {
+  let writes = [];
   const writeJSON = obj => {
     const jsonObj = serializeSlogObj(obj);
     // Prepend a newline so that the file can be more easily manipulated.
     const data = new TextEncoder().encode(`\n${jsonObj}`);
     // console.log('have obj', obj, data);
-    writeCircBuf(data);
+    writes.push(writeCircBuf(data));
```

Comment:
Since each write will update the arena you must make the writes sequential, aka chain the write to the completion of the previous promise. As such you don't keep an array of promise, but update single binding holding the latest promise.

Reply:
of course, thanks